stock_lot_suffix/models/stock_picking.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _set_channel_category(self):
+        for picking in self:
+            lots = self.move_line_ids.lot_id
+            if lots:
```

Comment:
Reduce the `if` ladder.

```suggestion
            if not lots:
                continue
```